src/Subscriber/AutoUpdateTitleWithLabelSubscriber.php

```diff
@@ -69,10 +72,22 @@ public function onPullRequest(GitHubEvent $event)
         }
 
         $repository = $event->getRepository();
-        $prNumber = $data['number'];
-        $this->pullRequestApi->update($repository->getVendor(), $repository->getName(), $prNumber, ['title' => $prTitle]);
+        $number = $data['number'];
+
+        $lock = $this->lockFactory->createLock($repository->getFullName().'#'.$number);
+        $lock->acquire(true); // blocking
+        try {
+            // Refetch the current title just to make sure it has not changed
+            if ($prTitle === ($this->pullRequestApi->show($repository, $number)['title'] ?? '')) {
+                return;
+            }
+            $this->pullRequestApi->updateTitle($repository, $number, $prTitle);
+        } finally {
+            $lock->release();
```

Comment:
not needed, Symfony do it for you on `__destruct`

Reply:
Oh, Thank you